### ISSUE TITLE
Push lib to Maven Central/jCenter

### ISSUE THREAD

**Issue by davidvavra** [kolavar/android-support-v4-preferencefragment#13]
Now it's pain to include the library.

**Comment by easyaspi314** [comment]
You should first change the minSdkVersion to 4, not 7. Then have a look at this 
(I haven't tried it): [*Maven – Guide to uploading artifacts to the Central Repository*](http://maven.apache.org/guides/mini/guide-central-repository-upload.html)

**Comment by destil** [comment]
I would vote for jcenter instead of Maven Central. It's so much easier.

**Comment by easyaspi314** [comment]
yeah i agree i have no idea how to use Maven Central. jCenter gives you an easy way to do that.

**Comment by oinegue** [comment]
How do I install the lib without maven? I'm using Android Studio

**Comment by fjtorres** [comment]
Add 47deg repository to build.gradle:

maven { url 'http://clinker.47deg.com/nexus/content/repositories/releases'}
or
maven { url 'http://clinker.47deg.com/nexus/content/repositories/snapshots'}

**Comment by easyaspi314** [comment]
How I did it:

1. Copy the m2repository/com folder into

Windows: Open Explorer and type in the address bar:
   %USERPROFILE%/.m2

Mac/Linux:
~/.m2

- Linux: Make sure you show hidden folders.
- Mac: You need to use the Menu Bar > Go > Go to folder... and type
   ~/.m2

2. In your project build.gradle, find the **second** dependencies { }
section (at the end) and add
   mavenLocal()
After that, in your module's build.gradle, add to your dependencies:
   compile 'com.android.support:support-v4-preferencefragment:1.0.0'

How do I install the lib without maven? I'm using Android Studio

--
Reply to this email directly or view it on GitHub
<https://github.com/kolavar/android-support-v4-preferencefragment/issues/13#issuecomment-70508078>
.

**Comment by oinegue** [comment]
Thank you! I'm in Linux and I had to put the m2repository/com folder in ~/.m2/repository/
Now I can use the lib.

**Comment by pawelantczak** [comment]
Hello.
If you prefer or must use remote repo, feel free to use my nexus:
```gradle
    repositories {
        mavenCentral()
        maven { url "http://nexus.antczak.org/content/repositories/thirdparty/" }
    }
```
then:
```gradle
    compile 'com.android.support:support-v4-preferencefragment:1.0.0'
```

**Comment by easyaspi314** [comment]
Can someone set this to be compatible with Donut? I know it is a bit silly, but the title says v4 but the build files say v7.

**Comment by pserwylo** [comment]
I'm also hoping that this can be uploaded to mavenCentral or jcenter at some point. We are using it for F-Droid, and while we have a submodule of this repo building correctly, I'd like to be able to get a copy of the binary from mavenCentral/jcenter too (much more convenient for the majority of contributors).

**Comment by mateberlin** [comment]
+1

**Comment by slavafomin** [comment]
+1

**Comment by easyaspi314** [comment]
+1
@kolavar: Are you there?

**Comment by shkschneider** [comment]
Would like to propose a pull request for JitPack support once the pull request #11 will be merged.

**Comment by PaulWoitaschek** [comment]
+1 This would be very useful

**Comment by danpfe** [comment]
+1

**Comment by msegers** [comment]
+1

**Comment by DKMudrechenko** [comment]
+1

**Comment by codewing** [comment]
+1

**Comment by ticofab** [comment]
+1

**Comment by mhousser** [comment]
+1. Really no reason I shouldn't be able to use this library with a single Gradle line within Android Studio, just like every other library. Right now this library is still one of those 'annoying ones that needs to have its own folder and committed files in source control etc.'

**Comment by mvdan** [comment]
Status of this? +1

@kolavar if you don't have time to work on this library, have you thought about letting someone else do the work for you?

**Comment by shkschneider** [comment]
I got tired of copying files, I updated the code (Gradle, dimens.xml from AOSP as of 5.1.1 etc) and pushed this to JitPack, if it could be useful for anyone: https://github.com/shkschneider/android_PreferenceFragmentCompat

**Comment by mvdan** [comment]
Reping @kolavar - there are also pending PRs that need attention. If you don't have time to maintain the library, have you thought of handing over the project to someone else?

**Comment by shkschneider** [comment]
I'm done with this issue and this (abandonned) library -- won't publish into JitPack.
See https://github.com/shkschneider/android_PreferenceFragmentCompat/blob/master/README.md.

Use [com.android.support:support-v7](https://developer.android.com/topic/libraries/support-library/features.html#v7-preference) and [com.android.support:preference-v14](https://developer.android.com/topic/libraries/support-library/features.html#v14-preference).
1. Use a `android.support.v7.preference.PreferenceFragmentCompat` as fragment.
2. Set `<item name="preferenceTheme">@style/PreferenceThemeOverlay.v14.Material</item>` in your theme.
3. Set `colors preference_accent` and `preference_fallback_accent_color` to `accentColor`.
4. Create your layout with a `Toolbar` that includes something like:
   <pre>&lt;fragment
   android:name="com.example.SettingsActivity$SettingsFragment"
   android:tag="SettingsActivity$SettingsFragment"
   android:layout_width="match_parent"
   android:layout_height="match_parent" /&gt;</pre>

Source: https://github.com/Gericop/Android-Support-Preference-V7-Fix.
